kore/src/main/scala/org/kframework/parser/TreeNodesToKORE.scala

```diff
@@ -24,18 +24,18 @@ class TreeNodesToKORE(parseSort: java.util.function.Function[String, Sort], stri
     case Ambiguity(items) => KApply(KLabel("amb"), KList(items.asScala.toList map apply asJava), Att.empty)
   }
 
-  def anonVar(sort: Sort): K = {
+  def anonVar(sort: Sort, t: Term): K = {
     val lbl = KLabel("#SemanticCastTo" + sort.toString())
-    if (strict) KApply(lbl, KList(KToken("_", Sorts.KVariable)), Att.empty.add(classOf[Production], Production(lbl, Seq(), sort, Seq(NonTerminal(sort, None))))) else KToken("_", Sorts.KVariable)
+    if (strict) KApply(lbl, KList(KToken("_", Sorts.KVariable, locationToAtt(t.location, t.source))), locationToAtt(t.location, t.source).add(classOf[Production], Production(lbl, Seq(), sort, Seq(NonTerminal(sort, None))))) else KToken("_", Sorts.KVariable, locationToAtt(t.location, t.source))
   }
 
   def termConsToKApply(t: TermCons): K = {
     if (t.production.att.contains("recordPrd", classOf[Production])) {
       val realProd = t.production.att.get("recordPrd", classOf[Production])
       val map = new util.ArrayList(t.items).asScala.reverse.zipWithIndex.map { case (item, idx) => (t.production.nonterminal(idx).name.get, apply(item))} toMap
       val realItems = realProd.nonterminals.map {
-        case NonTerminal(sort, None) => anonVar(sort)
-        case NonTerminal(sort, Some(x)) => map.getOrElse(x, anonVar(sort))
+        case NonTerminal(sort, None) => anonVar(sort, t)
+        case NonTerminal(sort, Some(x)) => map.getOrElse(x, anonVar(sort, t))
```

Comment:
It's a little confusing to me that you're passing the entire `t: Term` into `anonVar`, because why would it need a term? Are you able to pass just the `Location` in for it to use? I'm approving anyway, but something to think about.

Reply:
I changed the signature of the method to t: HasLocation to make it clearer what is happening.